File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {

        int n = nums.length;
        int count = 0;
        int res = 0;

        for(int i = 0; i < n; i++){
            if(nums[i] == 1){
                count++;
            }
            if(nums[i] == 0){
                count = 0;
            }
            if(count > res){
                res = count;
            }
        }
        return res;













    //     int res = 0;
    //     int count = 0;

    //     for (int n : nums) {
    //         if (n == 0) {
    //             count = 0;
    //         } else {
    //             count++;
    //         }

    //         if (res < count) {
    //             res = count;
        //     }
        // }

        // return res;        
    }
}

File: 485-max-consecutive-ones/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
        let n = nums.length;
        let count = 0;
        let res = 0;

        for(let i = 0; i < n; i++){
            if(nums[i] == 1){
                count++;
            }
            if(nums[i] == 0){
                count = 0;
            }
            if(count > res){
                res = count;
            }
        }
        return res;
};